File: Day3/Day3.java
// basic program 

import java.util.Scanner;

// Class Point
class Point {
    private int x;
    private int y;

    // Constructor
    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    // Getters
    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    // Setters
    public void setX(int x) {
        this.x = x;
    }

    public void setY(int y) {
        this.y = y;
    }

    // toString
    public String toString() {
        return "(" + x + ", " + y + ")";
    }

    //function pass by reference
    public void update(Point p) {
        p.setX(100);
        p.setY(200);
    }

    // int is primitive type
    int[] arr = new int[1];
    arr[0] = x;
    
    public void plug(int[] arr) {
        arr[0]++;
    }

}

File: Day3/Day3.cpp
// basic program

#include <iostream>
using namespace std;

class Point
{
private:
    int x;
    int y;

public:
    // constructor & destructor

    Point(){
        cout << "Constructor called" << endl;
        x = 0;
        y = 0;
    }

    //Overloaded constructor
    Point(int x, int y)
    {
        cout << "Constructor called" << endl;
        this->x = x;
        this->y = y;
    }

    Point(int x) : Point(){
        cout << "Debug" << endl;
        this->x = x;
    }

    // destructor automatically called when the object is deleted
    ~Point()
    {
        cout << "Destructor called" << endl;
    }

    // setter & getter
    void setX(int x)
    {
        this->x = x;
    }

    void setY(int y)
    {
        this->y = y;
    }

    int getX()
    {
        return x;
    }

    int getY()
    {
        return y;
    }
    // Method 

    void toString()
    {
        cout << "x: " << x << " y: " << y << endl;
    }

    //function to update the point by passing the object - no recommended
    // void update(Point p){
    //     this->x = p.x;
    //     this->y = p.y;
    // }

    //function to update the point by passing the reference of the object - recommended
    void update(Point &p){
        this->x = p.x;
        this->y = p.y;
    }

    //function to update the point by passing the pointer of the object - recommended
    void update(Point * p)
    {
        this->x = p->x;
        this->y = p->y;
    }
};

int main()
{
    Point p1(10);
    p1.toString();
    return 0;
}
